src/patterns/sns-lambda.ts

```diff
@@ -0,0 +1,40 @@
+import { SnsEventSource } from "@aws-cdk/aws-lambda-event-sources";
+import { Topic } from "@aws-cdk/aws-sns";
+import { CfnOutput } from "@aws-cdk/core";
+import type { GuLambdaErrorPercentageMonitoringProps } from "../constructs/cloudwatch/lambda-alarms";
+import type { NoMonitoring } from "../constructs/cloudwatch/no-monitoring";
+import type { GuStack } from "../constructs/core";
+import type { GuFunctionProps } from "../constructs/lambda";
+import { GuLambdaFunction } from "../constructs/lambda";
+import { GuSnsTopic } from "../constructs/sns/sns-topic";
+
+interface ExistingSnsTopic {
+  logicalIdFromCloudFormation?: string; // Allows us to inherit an SNS topic which has already been created via a CloudFormation stack
+  externalTopicName?: string; // Allows us to link the lambda to an SNS topic owned by a different stack (or created outside of CloudFormation)
+}
+
+interface GuSnsLambdaProps extends Omit<GuFunctionProps, "rules" | "apis" | "errorPercentageMonitoring"> {
+  monitoringConfiguration: NoMonitoring | GuLambdaErrorPercentageMonitoringProps;
+  existingSnsTopic?: ExistingSnsTopic;
+}
+
+export class GuSnsLambda extends GuLambdaFunction {
```

Comment:
As patterns are the primary entry point to the library, should we add documentation to them by default to make the typedoc site friendly?

Reply:
Yes, definitely! Thanks for the reminder!

Reply:
First pass at docs is here: https://github.com/guardian/cdk/pull/215/commits/80058bc3d1846b33dd6c46fa334f8bba3f204973